libs/local_lcos/include/hpx/local_lcos.hpp

```diff
@@ -0,0 +1,33 @@
+//  Copyright (c) 2007-2016 Hartmut Kaiser
+//  Copyright (c) 2011 Bryce Adelstein-Lelbach
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(HPX_6EB418B5_DC41_45A3_ADF4_C45A068F73D4)
+#define HPX_6EB418B5_DC41_45A3_ADF4_C45A068F73D4
+
+#include <hpx/config.hpp>
+#include <hpx/dataflow.hpp>
+#include <hpx/local_lcos/barrier.hpp>
+#include <hpx/local_lcos/channel.hpp>
+#include <hpx/local_lcos/condition_variable.hpp>
+#include <hpx/local_lcos/counting_semaphore.hpp>
+#include <hpx/local_lcos/event.hpp>
+#include <hpx/local_lcos/latch.hpp>
+#include <hpx/local_lcos/mutex.hpp>
+#include <hpx/local_lcos/no_mutex.hpp>
+#include <hpx/local_lcos/recursive_mutex.hpp>
+#include <hpx/local_lcos/shared_mutex.hpp>
+#include <hpx/local_lcos/sliding_semaphore.hpp>
+
+#include <hpx/lcos/future.hpp>
+#include <hpx/local_lcos/and_gate.hpp>
+#include <hpx/local_lcos/packaged_continuation.hpp>
+#include <hpx/local_lcos/packaged_task.hpp>
+#include <hpx/local_lcos/promise.hpp>
+#include <hpx/local_lcos/receive_buffer.hpp>
+#include <hpx/local_lcos/trigger.hpp>
+
+#endif
```

Comment:
I think we can generate this file now, but keep the original `hpx/include/local_lcos.hpp` in place. So far we've kept all the `hpx/include/*` headers in place in the main include directory. I think once we've moved everything we can go and deprecate that directory separately.